### PR TITLE
Injectable config

### LINKED ISSUE #14: extract hardcoded config params (not passed by profile) to struct and leave to caller to decide
```
adapter/library.cpp:91

        config.info = true;
        config.clockTickMS = 1000;   //ticks every 1 sec
        config.disableClientCert = true;  //we don't use certs for client identification
        config.connTimeout = 10; // connection timeout - 10 seconds (make it configurable?)
        config.tunPersist = true;
        config.compressionMode = "yes";
```

### PR DESCRIPTION
**Pull request by Waldz**
Closes #14

### CODE REVIEW

File: openvpn3/session.go [hunk 1]
```diff
@@ -88,37 +95,28 @@ var ErrInitFailed = errors.New("openvpn3 init failed")
...
+		defer cConfigUnregister()
 
-		cPassword := newCharPointer(creds.Password)
-		defer cPassword.delete()
-
-		cCreds := expCredentials{
-			username: cUsername.Ptr,
-			password: cPassword.Ptr,
-		}
+		cCrediantials, cCrediantialsUnregister := session.userCredentials.toPtr()
```
Comment (soffokl) [inline openvpn3/session.go]: typo here `Crediantials` -> `Credentials`

File: examples/desktop/main.go [hunk 2]
```diff
@@ -73,20 +73,15 @@ func main() {
 
 	openvpn3.SelfCheck(logger)
 
-	session := openvpn3.NewSession(&loggingCallbacks{})
-
 	bytes, err := ioutil.ReadFile(profileName)
```
Comment (soffokl) [inline examples/desktop/main.go]: `profileName` comes from `os.Args[1]` and panics if now arguments were passed.

File: openvpn3/session.go (hunk 1 above)
Comment (Waldz) [reply openvpn3/session.go]: Fixed

File: examples/desktop/main.go (hunk 2 above)
Comment (Waldz) [reply examples/desktop/main.go]: Fixed

**Pull request merged**